llama_index/indices/base.py

```diff
@@ -59,6 +61,7 @@ def __init__(
         self._service_context = service_context or ServiceContext.from_defaults()
         self._storage_context = storage_context or StorageContext.from_defaults()
         self._docstore = self._storage_context.docstore
+        self._show_progress = show_progress
```

Comment:
curious: what's the tradeoff for keeping this as an attribute instead of passing it directly into `build_index_from_nodes`.

From what I see:
* pro: avoid having to hack a bunch of function signatures to include show progress 
* con: keeping track of yet another configuration in the index object, less explicit which functions what this config affects.

Reply:
yea the main thing is it reduces the number of functions that have to accept the this attribute. I initially had it as an attribute on `build_index_from_nodes` but then realized that there are a ton of places that would have to get modified.